apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -0,0 +1,1234 @@
+	 spend_test_contract/1,
+         null/1
+        ]).
+
+-define(NODE, dev1).
+-define(DEFAULT_TESTS_COUNT, 5).
+-define(WS, aehttp_ws_test_utils).
+
+%% Contract return values for some predefined numbers. Conversion
+%% functions will make this unnecessary.
+
+-define(VALUE_STATEP, <<"0x0000000000000000000000000000000000000000000000000000000000000020">>).
+
+all() ->
+    [
+     {group, contracts}
+    ].
+
+groups() ->
+    [
+     {contracts, [sequence],
+      [
+       spending_1,
+       spending_2,
+       spending_3,
+       identity_contract,
+       simple_storage_contract,
+       counter_contract,
+       maps_contract,
+       environment_contract,
+       spend_test_contract,
+       dutch_auction_contract,
+       null                                     %This allows to end with ,
+      ]}
+    ].
+
+suite() ->
+    [].
+
+init_per_suite(Config) ->
+    ok = application:ensure_started(erlexec),
+    DataDir = ?config(data_dir, Config),
+    TopDir = aecore_suite_utils:top_dir(DataDir),
+    Config1 = [{symlink_name, "latest.http_contracts"},
+               {top_dir, TopDir},
+               {test_module, ?MODULE}] ++ Config,
+    aecore_suite_utils:make_shortcut(Config1),
+    ct:log("Environment = ~p", [[{args, init:get_arguments()},
+                                 {node, node()},
+                                 {cookie, erlang:get_cookie()}]]),
+    Forks = aecore_suite_utils:forks(),
+
+    aecore_suite_utils:create_configs(Config1, #{<<"chain">> =>
+                                                 #{<<"persist">> => true,
+                                                   <<"hard_forks">> => Forks}}),
+    aecore_suite_utils:make_multi(Config1, [?NODE]),
+    [{nodes, [aecore_suite_utils:node_tuple(?NODE)]}]  ++ Config1.
+
+end_per_suite(_Config) ->
+    ok.
+
+init_per_group(contracts, Config) ->
+    NodeName = aecore_suite_utils:node_name(?NODE),
+    aecore_suite_utils:start_node(?NODE, Config),
+    aecore_suite_utils:connect(NodeName),
+
+    {ok, _} = aecore_suite_utils:mine_blocks(NodeName, 2),
+
+    %% Prepare accounts, Alice, Bert, Carl and Diana.
+
+    StartAmt = 5000,
+    {APubkey, APrivkey} = new_account(StartAmt),
+    {BPubkey, BPrivkey} = new_account(StartAmt),
+    {CPubkey, CPrivkey} = new_account(StartAmt),
+    {DPubkey, DPrivkey} = new_account(StartAmt),
+
+    {ok, [Block|_]} = aecore_suite_utils:mine_blocks(NodeName, 2),
+    ct:pal("Block ~p\n", [aec_blocks:txs(Block)]),
+
+    %% Save account information.
+    Accounts = #{acc_a => #{pub_key => APubkey,
+			    priv_key => APrivkey,
+			    start_amt => StartAmt},
+		 acc_b => #{pub_key => BPubkey,
+			    priv_key => BPrivkey,
+			    start_amt => StartAmt},
+		 acc_c => #{pub_key => CPubkey,
+			    priv_key => CPrivkey,
+			    start_amt => StartAmt},
+		 acc_d => #{pub_key => DPubkey,
+			    priv_key => DPrivkey,
+			    start_amt => StartAmt}},
+    [{accounts,Accounts},{node_name,NodeName}|Config];
```

Comment:
It seems that this init does not guarantee that Alice/Bert/Carl/Diana have StartAmt tokens, as it puts one spend tx per account, having a sender the mining node, in the mempool though without checking inclusion in block. (This is neither good nor bad on its own - it depends what test case assumes.)

Reply:
I have converted it to behave in exactly the same way is in the integration_SUITE.